src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,28 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    private LocalDateTime date;
+
+
+    public Deadline(String description, LocalDateTime date) {
+        super(description);
+        this.date = date;
+    }
+
+    @Override
+    public String getTaskType() {
+        return "D";
+    }
+
+    private String formatDate() {
+        return date.format(DateTimeFormatter.ofPattern("MMM d yyyy h:mm a"));
+    }
+
+    @Override
+    public String toString() {
+        return "[" + getTaskType() + "]" + "[" + getStatusIcon() + "]" + description + " " + "(by: " + formatDate() + ")";
```

Comment:
Perhaps you might want to break this line for better readability(Too many characters):
        `return "[" + getTaskType() + "]" + "[" + getStatusIcon() + "]" `
              `+ description + " " + "(by: " + formatDate() + ")";`